optuna/integration/catalyst.py

```diff
@@ -1,49 +1,38 @@
-from typing import Any
+from packaging import version
 
-import optuna
-from optuna._experimental import experimental
+from optuna._deprecated import deprecated
 from optuna._imports import try_import
 
 
 with try_import() as _imports:
-    from catalyst.dl import Callback
+    import catalyst
+
+    if version.parse(catalyst.__version__) < version.parse("21.3"):
+        raise ImportError(
+            f"You don't have Catalyst 21.3 installed! Catalyst version: {catalyst.__version__}"
+        )
+    from catalyst.dl import OptunaPruningCallback
 
 if not _imports.is_successful():
-    Callback = object  # NOQA
+    OptunaPruningCallback = object  # NOQA
 
 
-@experimental("2.0.0")
-class CatalystPruningCallback(Callback):
+@deprecated("2.7.0")
+class CatalystPruningCallback(OptunaPruningCallback):  # type: ignore
     """Catalyst callback to prune unpromising trials.
 
-    See `the example <https://github.com/optuna/optuna/blob/master/
-    examples/catalyst_simple.py>`_ if you want to add a pruning callback
-    which observes the accuracy of Catalyst's ``SupervisedRunner``.
+
+    .. seealso::
+
+        See https://catalyst-team.github.io/catalyst/api/callbacks.html?highlight=optuna#catalyst.callbacks.optuna.OptunaPruningCallback
+        for the detailed documentation.
```

Comment:
Users may not understand why they need to see the Catalyst's document. How about explaining the relationship between `CatalystPruningCallback` and `OptunaPruningCallback`?